### PR TITLE
Add bindings support to meeting start / end dates

### PR DESCRIPTION
**Pull request by melohagan**
## Description
Realised that the calendar invite feature is much more useful with bindings.

Also removed code to complete a deprecation for the Zapier and Make automation steps.

I initially had the `Use values / use bindings` toggle but it was a bit overkill for just the two date fields. That's why I added in the `DrawerBindableSlot` which allows you to use bindings directly with the date picker (or any other field in theory).

If the date is not valid, then the input will become text, e.g. handlebars or JavaScript bindings. When a valid date is entered it will automatically switch back to the calendar.

If in "text input mode", clicking on the Cancel icon will clear the field, and switch to calendar mode. See the video below!

Decided to disable the clear button for JavaScript bindings because you can't see what you're clearing, it just says '(JavaScript binding)', and it could potentially be a lot of code that gets accidentally wiped.

Boolean fields in the row selector have also been updated to use a select with three options, and tested for a bug fix around the update row action step:
- Choose an option (will ignore the field if not set)
- True
- False

Addresses: 
- https://linear.app/budibase/issue/SUP-101/add-bindings-to-meeting-startend-dates
- https://github.com/Budibase/budibase/issues/9764

## Screenshots
https://github.com/Budibase/budibase/assets/101575380/f43837bf-241b-4a6e-a9e0-9937436b2438

js

*Field with JavaScript binding and disabled clear button*

<br />
Screenshot 2023-09-15 at 18 05 31

*Date field with picker*

<br />
handlebars

*Field with editable handlebars text and clear button*

bool

*Boolean now has three options to allow it to be ignored*

### Removed the Use values / use bindings dropdown
Screenshot 2023-09-25 at 18 40 34

Screenshot 2023-09-25 at 18 41 47

## Documentation
- [x] https://linear.app/budibase/issue/SUP-71/add-send-meeting-invite-via-email-smtp-docs

### CODE REVIEW

**Comment by melohagan** [comment]
Backwards compatibility test:

Screenshot 2023-09-25 at 21 54 06

Screenshot 2023-09-25 at 21 50 18

This is correct behaviour because of the linked bug. Up to now the false toggle has been considered as *ignore*. You can now set this as *False* to overwrite the existing checkbox.

File: packages/builder/src/components/common/bindings/DrawerBindableSlot.svelte [hunk 1]
```diff
@@ -0,0 +1,254 @@
...
+      return !value
+    }
+    return links.every(link => schema?.constraints?.inclusion?.includes(link))
+  }
+
+  const isValidBoolean = value => {
+    return value === "false" || value === "true" || value == ""
+  }
+
+  const isValid = value => {
```
Comment (PClmnt) [inline packages/builder/src/components/common/bindings/DrawerBindableSlot.svelte:94]: for this function could we do something like:

```
const validationMap = {
  date: isValidDate,
  link: hasValidLinks,
  array: hasValidOptions,
  longform: value => !isJSBinding(value),
  json: value => !isJSBinding(value),
  boolean: isValidBoolean,
};

const isValid = value => {
  const validate = validationMap[type];
  return validate ? validate(value) : true;
};


```
to make it bit more succinct?

**Review by PClmnt — APPROVED** [review]
Just one small suggestion but looks good!!

File: packages/builder/src/components/common/bindings/DrawerBindableSlot.svelte (hunk 1 above)
Comment (melohagan) [reply packages/builder/src/components/common/bindings/DrawerBindableSlot.svelte:94]: Nice - I like it!

**Pull request merged**